packages/app/src/components/deposit/WithdrawModal.tsx

```diff
@@ -0,0 +1,273 @@
+                          </TableCell>
+                          <TableCell border="bottom">
+                            <Text>{blockTimestampToDate(lock.timestamp).toLocaleDateString("en-US")}</Text>
+                          </TableCell>
+                          <TableCell border="bottom">
+                            <Text>{getUnlockDateForProtocol(protocolName, lock, lastGovUnstakeEvent)}</Text>
+                          </TableCell>
+                          <TableCell border="bottom">
+                            <Button
+                              primary
+                              disabled={lock.open === false}
+                              style={{ padding: "5px 5px" }}
+                              onClick={async (e) => {
+                                e.preventDefault();
+                                await withdraw(lock.unstakeLockID);
+                                onDismiss();
+                              }}
+                              label="Withdraw"
+                            />
+                          </TableCell>
+                        </TableRow>
+                      );
+                    })}
+                  </TableBody>
+                </Table>
+              </Box>
+            </CardBody>
+            <CardFooter justify="center" pad={{ vertical: "medium" }}></CardFooter>
+          </Card>
+        </Layer>
+      )}
+    </>
+  );
+};
+
+export default WithdrawModal;
+
+const getUnlockDateForProtocol = (
+  protocolName: ProtocolName,
+  lock: Lock,
+  govUnstakeEvent: Queries.LastGovernanceUnstake | undefined
+): string => {
+  if (lock.open) {
+    return "Ready";
+  }
+  switch (protocolName) {
+    case "graph": {
+      if (govUnstakeEvent != null) {
+        const daysSinceLastGovUnstake = daysBetweenBlockTimestamps(
+          govUnstakeEvent.unstakeEvents[0].timestamp,
+          lock.timestamp
+        );
+        if (lock.unstakeLockID < govUnstakeEvent.unstakeEvents[0].unstakeLockID) {
+          return `${28 - daysSinceLastGovUnstake} days remaining`;
+        } else {
+          return `${28 + 28 - daysSinceLastGovUnstake} days remaining`;
+        }
+      } else {
+        return "Loading...";
+      }
+    }
+    case "audius": {
+      const unstakeDate = blockTimestampToDate(lock.timestamp);
+      const unlockDate = new Date(unstakeDate);
+      unlockDate.setDate(unstakeDate.getDate() + 7);
+      const daysUntilUnlock = daysBetweenDates(unstakeDate, unlockDate);
+      if (daysUntilUnlock < 0) {
+        return "Ready";
+      } else {
+        return `${daysUntilUnlock === 0 ? 1 : daysUntilUnlock} days remaining`;
+      }
+    }
+    case "livepeer": {
+      const unstakeDate = blockTimestampToDate(lock.timestamp);
+      const unlockDate = new Date(unstakeDate);
+      unlockDate.setDate(unstakeDate.getDate() + 7);
+      const daysUntilUnlock = daysBetweenDates(unstakeDate, unlockDate);
+      if (daysUntilUnlock < 0) {
+        return "Ready";
```

Comment:
it wasn't ditched, we are using it here: https://github.com/Tenderize/tender-app/pull/343/files#diff-a2b4930563c56927d6bb67b4e795d71c190bd48acb1e7e949d19d20b309dab34R80

we do an early return in case it's already available. (line 203)

Reply:
Ahh I see; what about the 0 Days/Ready thing? On the last day of the bonding period, I mean

Reply:
there's no such thing as 0 days, also we are not really caring about hourly deadline, so it's safer to assume it's 1 day

Reply:
I don't really think I agree here, I think we should switch to more granular time displays

- `>= 24 hours -> days `
- `> 1 hour <= 24 hours -> hours`
- `< 1 hour -> minutes`

Can do it in a follow up PR though